src/adagrams.js

```diff
@@ -1,15 +1,131 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  // initialize object to hold letter frequency table
+  const letterFreqs = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+
+  // fill letterPool array with all the letters
+  const letterPool = [];
+
+  for (const [key, value] of Object.entries(letterFreqs)) {
+    for (let i = 1; i < value + 1; i++) {
+      letterPool.push(key);
+    }
+  }
+
+  // draw letters into hand array
+  let hand = [];
+
+  for (let i = 0; i < 10; i++) {
+    let selector = Math.random() * letterPool.length;
+
+    let letter = letterPool.splice(selector, 1)[0];
+
+    hand.push(letter);
+  }
+
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let handCopy = lettersInHand;
```

Comment:
Not sure if you were trying to make a shallow copy here in an effort to avoid mutation, but just note that any mutation applied to `handCopy` will also apply to `lettersInHand`. If you want a shallow copy, you can instead use the spread operator:
```javascript
const handCopy = [...lettersInHand];
handCopy.pop()
// now both handCopy and lettersInHand would be missing their last element.
```
Also, `handCopy` should be a `const`